benchmarks/benchmarks/benchmark_algorithms.py

```diff
@@ -50,6 +50,37 @@ def time_average_clustering(self, graph):
         _ = nx.average_clustering(self.graphs_dict[graph])
 
 
+class DirectedAlgorithmBenchmarks:
+    timeout = 120
+    _graphs = [
+        nx.erdos_renyi_graph(100, 0.1, directed=True),
+        nx.erdos_renyi_graph(100, 0.5, directed=True),
+        nx.erdos_renyi_graph(100, 0.9, directed=True),
+        nx.erdos_renyi_graph(1000, 0.01, directed=True),
+        nx.erdos_renyi_graph(1000, 0.05, directed=True),
+        nx.erdos_renyi_graph(1000, 0.09, directed=True),
+    ]
+    params = [
+        "Erdos Renyi (100, 0.1)",
+        "Erdos Renyi (100, 0.5)",
+        "Erdos Renyi (100, 0.9)",
+        "Erdos Renyi (1000, 0.01)",
+        "Erdos Renyi (1000, 0.05)",
+        "Erdos Renyi (1000, 0.09)",
+    ]
+
+    param_names = ["graph"]
+
+    def setup(self, graph):
+        self.graphs_dict = dict(zip(self.params, self._graphs))
+
+    def time_tarjan_scc(self, graph):
+        _ = nx.strongly_connected_components(self.graphs_dict[graph])
+
+    def time_kosaraju_scc(self, graph):
+        _ = nx.kosaraju_strongly_connected_components(self.graphs_dict[graph])
```

Comment:
These functions actually return generators, so as it stands these benchmarks only test the generator creation time. To probe the functionality of the generators, they need to be iterated over:

```suggestion
    def time_tarjan_scc(self, graph):
        _ = list(nx.strongly_connected_components(self.graphs_dict[graph]))

    def time_kosaraju_scc(self, graph):
        _ = list(nx.kosaraju_strongly_connected_components(self.graphs_dict[graph]))
```